File: ts/1792/1688.ts
// 模拟
function numberOfMatches(n: number): number {
  let sum = 0
  while (n > 1) {
    n /= 2
    sum += Math.floor(n)
    n = Math.ceil(n)
  }
  return sum
};

// 数学
function numberOfMatches1(n: number): number {
  return n - 1
};

File: c/1792/1688.c
// 模拟
int numberOfMatches(int n){
  int sum = 0;
  while (n > 1) {
    if (n % 2) {
      sum += (n - 1) / 2;
      n += 1;
    } else {
      sum += n / 2;
    }
    n /= 2;
  }
  return sum;
}

// 数学
int numberOfMatches1(int n){
  return n - 1;
}